### ISSUE TITLE
PHP 7.0 compatibility

### ISSUE THREAD

**Issue by thedotedge** [php/pecl-numbers-bitset#10]
```
downloading bitset-2.0.1.tgz ...
Starting to download bitset-2.0.1.tgz (19,436 bytes)
......done: 19,436 bytes
5 source files, building
running: phpize
Configuring for:
PHP Api Version:         20151012
Zend Module Api No:      20151012
Zend Extension Api No:   320151012
building in /private/tmp/pear/temp/pear-build-rootu9VGkb/bitset-2.0.1
running: /private/tmp/pear/temp/bitset/configure --with-php-config=/usr/local/php5/bin/php-config
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for a sed that does not truncate output... /usr/bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-apple-darwin15.0.0
checking host system type... x86_64-apple-darwin15.0.0
checking target system type... x86_64-apple-darwin15.0.0
checking for PHP prefix... /usr/local/php5
checking for PHP includes... -I/usr/local/php5/include/php -I/usr/local/php5/include/php/main -I/usr/local/php5/include/php/TSRM -I/usr/local/php5/include/php/Zend -I/usr/local/php5/include/php/ext -I/usr/local/php5/include/php/ext/date/lib
checking for PHP extension directory... /usr/local/php5/lib/php/extensions/no-debug-non-zts-20151012
checking for PHP installed headers prefix... /usr/local/php5/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
checking for gawk... no
checking for nawk... no
checking for awk... awk
checking if awk is broken... no
checking whether to enable bitset support... yes, shared
checking for ld used by cc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /usr/bin/nm
checking whether ln -s works... yes
checking how to recognize dependent libraries... pass_all
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking the maximum length of command line arguments... 196608
checking command to parse /usr/bin/nm output from cc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking if cc supports -fno-rtti -fno-exceptions... yes
checking for cc option to produce PIC... -fno-common
checking if cc PIC flag -fno-common works... yes
checking if cc static flag -static works... no
checking if cc supports -c -o file.o... yes
checking whether the cc linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin15.0.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no

creating libtool
appending configuration tag "CXX" to libtool
configure: creating ./config.status
config.status: creating config.h
running: make
/bin/sh /private/tmp/pear/temp/pear-build-rootu9VGkb/bitset-2.0.1/libtool --mode=compile cc  -I. -I/private/tmp/pear/temp/bitset -DPHP_ATOM_INC -I/private/tmp/pear/temp/pear-build-rootu9VGkb/bitset-2.0.1/include -I/private/tmp/pear/temp/pear-build-rootu9VGkb/bitset-2.0.1/main -I/private/tmp/pear/temp/bitset -I/usr/local/php5/include/php -I/usr/local/php5/include/php/main -I/usr/local/php5/include/php/TSRM -I/usr/local/php5/include/php/Zend -I/usr/local/php5/include/php/ext -I/usr/local/php5/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2 -Wall   -c /private/tmp/pear/temp/bitset/bitset.c -o bitset.lo
mkdir .libs
 cc -I. -I/private/tmp/pear/temp/bitset -DPHP_ATOM_INC -I/private/tmp/pear/temp/pear-build-rootu9VGkb/bitset-2.0.1/include -I/private/tmp/pear/temp/pear-build-rootu9VGkb/bitset-2.0.1/main -I/private/tmp/pear/temp/bitset -I/usr/local/php5/include/php -I/usr/local/php5/include/php/main -I/usr/local/php5/include/php/TSRM -I/usr/local/php5/include/php/Zend -I/usr/local/php5/include/php/ext -I/usr/local/php5/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -Wall -c /private/tmp/pear/temp/bitset/bitset.c  -fno-common -DPIC -o .libs/bitset.o
In file included from /private/tmp/pear/temp/bitset/bitset.c:31:
/private/tmp/pear/temp/bitset/php_bitset.h:44:2: error: unknown type name 'zend_object_handle'; did you mean 'zend_object_handlers'?
        zend_object_handle handle;
        ^~~~~~~~~~~~~~~~~~
        zend_object_handlers
/usr/local/php5/include/php/Zend/zend_types.h:79:38: note: 'zend_object_handlers' declared here
typedef struct _zend_object_handlers zend_object_handlers;
                                     ^
/private/tmp/pear/temp/bitset/bitset.c:39:8: error: unknown type name 'zend_object_value'
static zend_object_value bitset_objects_new(zend_class_entry *ce TSRMLS_DC);
       ^
/private/tmp/pear/temp/bitset/bitset.c:40:8: error: unknown type name 'zend_object_value'
static zend_object_value php_bitset_register_object(php_bitset_object *intern TSRMLS_DC);
       ^
/private/tmp/pear/temp/bitset/bitset.c:424:67: error: too many arguments provided to function-like macro invocation
                RETURN_STRINGL((char *) intern->bitset_val, intern->bitset_len, 1);
                                                                                ^
/usr/local/php5/include/php/Zend/zend_API.h:638:9: note: macro 'RETURN_STRINGL' defined here
#define RETURN_STRINGL(s, l)                    { RETVAL_STRINGL(s, l); return; }
        ^
/private/tmp/pear/temp/bitset/bitset.c:424:3: error: use of undeclared identifier 'RETURN_STRINGL'
                RETURN_STRINGL((char *) intern->bitset_val, intern->bitset_len, 1);
                ^
/private/tmp/pear/temp/bitset/bitset.c:738:16: error: no member named 'type' in 'struct _zval_struct'
        return_value->type = IS_OBJECT;
        ~~~~~~~~~~~~  ^
/private/tmp/pear/temp/bitset/bitset.c:777:16: error: no member named 'type' in 'struct _zval_struct'
        return_value->type = IS_OBJECT;
        ~~~~~~~~~~~~  ^
/private/tmp/pear/temp/bitset/bitset.c:789:59: error: too many arguments provided to function-like macro invocation
        while (zend_hash_get_current_data(Z_ARRVAL_P(bit_array), (void **) &entry) == SUCCESS) {
                                                                 ^
/usr/local/php5/include/php/Zend/zend_hash.h:187:9: note: macro 'zend_hash_get_current_data' defined here
#define zend_hash_get_current_data(ht) \
        ^
/private/tmp/pear/temp/bitset/bitset.c:789:9: error: use of undeclared identifier 'zend_hash_get_current_data'; did you mean 'zend_hash_get_current_data_ex'?
        while (zend_hash_get_current_data(Z_ARRVAL_P(bit_array), (void **) &entry) == SUCCESS) {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~
               zend_hash_get_current_data_ex
/usr/local/php5/include/php/Zend/zend_hash.h:171:30: note: 'zend_hash_get_current_data_ex' declared here
ZEND_API zval* ZEND_FASTCALL zend_hash_get_current_data_ex(HashTable *ht, HashPosition *pos);
                             ^
/private/tmp/pear/temp/bitset/bitset.c:789:9: warning: comparison of function 'zend_hash_get_current_data_ex' equal to a null pointer is always false [-Wtautological-pointer-compare]
        while (zend_hash_get_current_data(Z_ARRVAL_P(bit_array), (void **) &entry) == SUCCESS) {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~                                             ~~~~~~~
/private/tmp/pear/temp/bitset/bitset.c:789:9: note: prefix with the address-of operator to silence this warning
        while (zend_hash_get_current_data(Z_ARRVAL_P(bit_array), (void **) &entry) == SUCCESS) {
               ^
               &
/private/tmp/pear/temp/bitset/bitset.c:790:7: warning: implicit declaration of function 'Z_TYPE_PP' is invalid in C99 [-Wimplicit-function-declaration]
                if (Z_TYPE_PP(entry) != IS_LONG) {
                    ^
/private/tmp/pear/temp/bitset/bitset.c:791:4: warning: incompatible pointer types initializing 'zval *' (aka 'struct _zval_struct *') with an expression of type 'zval **' (aka 'struct _zval_struct **'); dereference with * [-Wincompatible-pointer-types]
                        SEPARATE_ZVAL(entry);
                        ^~~~~~~~~~~~~~~~~~~~
/usr/local/php5/include/php/Zend/zend_types.h:948:9: note: expanded from macro 'SEPARATE_ZVAL'
                zval *_zv = (zv);                                                               \
                      ^     ~~~~
/private/tmp/pear/temp/bitset/bitset.c:795:7: warning: implicit declaration of function 'Z_LVAL_PP' is invalid in C99 [-Wimplicit-function-declaration]
                if (Z_LVAL_PP(entry) > 0) {
                    ^
/private/tmp/pear/temp/bitset/bitset.c:871:39: error: too many arguments provided to function-like macro invocation
                RETURN_STRINGL((char *)retval, len, 0);
                                                    ^
/usr/local/php5/include/php/Zend/zend_API.h:638:9: note: macro 'RETURN_STRINGL' defined here
#define RETURN_STRINGL(s, l)                    { RETVAL_STRINGL(s, l); return; }
        ^
/private/tmp/pear/temp/bitset/bitset.c:871:3: error: use of undeclared identifier 'RETURN_STRINGL'
                RETURN_STRINGL((char *)retval, len, 0);
                ^
/private/tmp/pear/temp/bitset/bitset.c:909:33: warning: implicit declaration of function 'zend_object_store_get_object' is invalid in C99 [-Wimplicit-function-declaration]
        intern = (php_bitset_object *) zend_object_store_get_object(object TSRMLS_CC);
                                       ^
/private/tmp/pear/temp/bitset/bitset.c:909:11: warning: cast to 'php_bitset_object *' from smaller integer type 'int' [-Wint-to-pointer-cast]
        intern = (php_bitset_object *) zend_object_store_get_object(object TSRMLS_CC);
                 ^
/private/tmp/pear/temp/bitset/bitset.c:938:53: error: too many arguments provided to function-like macro invocation
        while (zend_hash_get_current_data(Z_ARRVAL_P(arr), (void **) &entry) == SUCCESS) {
                                                           ^
/usr/local/php5/include/php/Zend/zend_hash.h:187:9: note: macro 'zend_hash_get_current_data' defined here
#define zend_hash_get_current_data(ht) \
        ^
/private/tmp/pear/temp/bitset/bitset.c:938:9: error: use of undeclared identifier 'zend_hash_get_current_data'; did you mean 'zend_hash_get_current_data_ex'?
        while (zend_hash_get_current_data(Z_ARRVAL_P(arr), (void **) &entry) == SUCCESS) {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~
               zend_hash_get_current_data_ex
/usr/local/php5/include/php/Zend/zend_hash.h:171:30: note: 'zend_hash_get_current_data_ex' declared here
ZEND_API zval* ZEND_FASTCALL zend_hash_get_current_data_ex(HashTable *ht, HashPosition *pos);
                             ^
/private/tmp/pear/temp/bitset/bitset.c:938:9: warning: comparison of function 'zend_hash_get_current_data_ex' equal to a null pointer is always false [-Wtautological-pointer-compare]
        while (zend_hash_get_current_data(Z_ARRVAL_P(arr), (void **) &entry) == SUCCESS) {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~                                       ~~~~~~~
/private/tmp/pear/temp/bitset/bitset.c:938:9: note: prefix with the address-of operator to silence this warning
        while (zend_hash_get_current_data(Z_ARRVAL_P(arr), (void **) &entry) == SUCCESS) {
               ^
               &
/private/tmp/pear/temp/bitset/bitset.c:940:4: warning: incompatible pointer types initializing 'zval *' (aka 'struct _zval_struct *') with an expression of type 'zval **' (aka 'struct _zval_struct **'); dereference with * [-Wincompatible-pointer-types]
                        SEPARATE_ZVAL(entry);
                        ^~~~~~~~~~~~~~~~~~~~
/usr/local/php5/include/php/Zend/zend_types.h:948:9: note: expanded from macro 'SEPARATE_ZVAL'
                zval *_zv = (zv);                                                               \
                      ^     ~~~~
/private/tmp/pear/temp/bitset/bitset.c:941:4: warning: incompatible pointer types passing 'zval **' (aka 'struct _zval_struct **') to parameter of type 'const zval *' (aka 'const struct _zval_struct *'); remove & [-Wincompatible-pointer-types]
                        convert_to_long_ex(entry);
                        ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/php5/include/php/Zend/zend_operators.h:408:34: note: expanded from macro 'convert_to_long_ex'
#define convert_to_long_ex(pzv)         convert_to_ex_master(pzv, long, IS_LONG)
                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/php5/include/php/Zend/zend_operators.h:368:6: note: expanded from macro 'convert_to_ex_master'
        if (Z_TYPE_P(pzv)!=upper_type) {                                        \
            ^~~~~~~~~~~~~
/usr/local/php5/include/php/Zend/zend_types.h:337:28: note: expanded from macro 'Z_TYPE_P'
#define Z_TYPE_P(zval_p)                        Z_TYPE(*(zval_p))
                                                ^~~~~~~~~~~~~~~~~
/usr/local/php5/include/php/Zend/zend_types.h:336:39: note: expanded from macro 'Z_TYPE'
#define Z_TYPE(zval)                            zval_get_type(&(zval))
                                                              ^~~~~~~
/usr/local/php5/include/php/Zend/zend_types.h:326:64: note: passing argument to parameter 'pz' here
static zend_always_inline zend_uchar zval_get_type(const zval* pz) {
                                                               ^
/private/tmp/pear/temp/bitset/bitset.c:941:23: warning: incompatible pointer types passing 'zval **' (aka 'struct _zval_struct **') to parameter of type 'zval *' (aka 'struct _zval_struct *'); dereference with * [-Wincompatible-pointer-types]
                        convert_to_long_ex(entry);
                                           ^~~~~
                                           *
/usr/local/php5/include/php/Zend/zend_operators.h:408:55: note: expanded from macro 'convert_to_long_ex'
#define convert_to_long_ex(pzv)         convert_to_ex_master(pzv, long, IS_LONG)
                                                             ^
/usr/local/php5/include/php/Zend/zend_operators.h:369:27: note: expanded from macro 'convert_to_ex_master'
                convert_to_##lower_type(pzv);                                           \
                                        ^
/usr/local/php5/include/php/Zend/zend_operators.h:244:51: note: passing argument to parameter 'op' here
ZEND_API void ZEND_FASTCALL convert_to_long(zval *op);
                                                  ^
/private/tmp/pear/temp/bitset/bitset.c:974:8: error: unknown type name 'zend_object_value'
static zend_object_value bitset_objects_new(zend_class_entry *ce TSRMLS_DC)
       ^
/private/tmp/pear/temp/bitset/bitset.c:985:8: error: unknown type name 'zend_object_value'
static zend_object_value php_bitset_register_object(php_bitset_object *intern TSRMLS_DC)
       ^
/private/tmp/pear/temp/bitset/bitset.c:987:19: error: expected ';' after expression
        zend_object_value retval;
                         ^
                         ;
/private/tmp/pear/temp/bitset/bitset.c:987:2: error: use of undeclared identifier 'zend_object_value'
        zend_object_value retval;
        ^
/private/tmp/pear/temp/bitset/bitset.c:987:20: error: use of undeclared identifier 'retval'
        zend_object_value retval;
                          ^
/private/tmp/pear/temp/bitset/bitset.c:988:2: error: use of undeclared identifier 'retval'
        retval.handle = zend_objects_store_put(intern, (zend_objects_store_dtor_t) zend_objects_destroy_object,  bitset_objects_free_storage, NULL TSRMLS_CC);
        ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
10 warnings and 20 errors generated.
make: *** [bitset.lo] Error 1
ERROR: `make' failed
```

**Comment by willfitch** [comment]
@thedotedge I'm updating this to 7.0 as we speak.  This will be released as 3.0.

**Comment by willfitch** [comment]
@thedotedge - please checkout the beta release at https://pecl.php.net/package/Bitset, or download the latest release from the https://github.com/php/pecl-numbers-bitset/releases for 3.0.0.  This includes updating to PHP 7.0.x
